docs/reference/index-modules/indexing-pressure.asciidoc

```diff
@@ -0,0 +1,56 @@
+[[index-modules-indexing-pressure]]
+== Indexing Pressure
+
+Indexing documents into Elasticsearch introduces system load in the form of
+memory and CPU load. Each indexing operation includes coordinating, primary, and
+replica stages. These stages can be performed across multiple nodes in the
+cluster. If too much indexing work is introduced into the system, the cluster
+can become saturated.
+
+Indexing pressure is primarily generated by external operations such as indexing
+requests or CCR follow tasks. However, there is some internal indexing pressure
+during a shard recovery or primary failover.
+
+Elasticsearch internally monitors indexing load. When the load exceeds
+certain limits, new indexing work will be rejected.
+
+[float]
+=== Indexing Stages
```

Comment:
This overlaps somewhat with the `Basic write model` section of `data-replication.asciidoc` ( which is mostly missing the coordinating bits). I wonder if we can extend that one and link to it here?